### PR TITLE
Update Builder dev doc / scripts

### PR DESCRIPTION
**Pull request by chefsalim**
This PR updates the Builder development documentation to clean up, simplify and tighten the steps that are required in order to configure the environment. It also updates the Web development readme that was significantly out of date.

Also there are a couple of small bug fixes/changes to the env scripts.

Signed-off-by: Salim Alam <salam@chef.io>

tenor-262902233

### CODE REVIEW

**Comment by thesentinels** [comment]
Thanks for the pull request! Here is what will happen next:
 1. Your PR will be reviewed by the maintainers
 2. If everything looks good, one of them will `approve` it, and your PR will be merged.

Thank you for contributing!

File: DEVELOPING.md [hunk 1]
```diff
@@ -1,210 +1,205 @@
...
+* *Ubuntu Desktop 18.04 LTS*. Other distributions should work, but this is the one we will support.
+
+* *Decision on where you want to run the Web UI*. If you are doing active Web UI development, then you will likely want to run the UI on your Host (Mac OS). If so, there are some extra steps and configuration changes that will be needed, and those are called out below. See the [Web UI README](https://github.com/habitat-sh/builder/blob/master/components/builder-web/README.md) for more info.
+
+## Host OS Provisioning
+
+If you plan to run the UI on the host (Mac) OS, you will need to ensure that the ports are forwarded appropriately.
+
+Add the following lines to `/Library/Preferences/VMware Fusion/networking`:
 ```
```
Comment (mgamini) [inline DEVELOPING.md]: To get your VM IP addr, enter a terminal in the vm, and find the address in the output from `ip address`

File: DEVELOPING.md [hunk 2]
```diff
@@ -1,210 +1,205 @@
...
-PGPASSWORD=$(cat /hab/svc/builder-datastore/config/pwfile) tools/merge-shards/merge-shards.sh originsrv migrate
-```
+### Starting Builder services
+Once the Builder Repo is configured, Builder services can be started inside the Habitat Studio.
 
-If all goes well, you'll see some messages about things being all good, and
-your services should behave normally. If they don't, exit the studio and
-re-enter.
+* `cd ${BUILDER_SRC_ROOT}`
+* `direnv allow`
```
Comment (mgamini) [inline DEVELOPING.md]: Is direnv installed by default? Maybe add `apt-get install direnv`

File: DEVELOPING.md [hunk 3]
```diff
@@ -1,210 +1,205 @@
...
+### Seeding base packages
 
-You may use the `load_package` helper to specify a package to upload. Ex:
+In order to do package builds locally, at a minimum you will need to seed the your dev repo with the latest version of `core/hab-backline`.
+
+From within your Studio, do the following (for example, using the 0.64.1 version of hab-backline):
 
 ```
-load_package /hab/cache/artifacts/core-*.hart
+load_package /hab/cache/artifacts/core-hab-backlin-0.64.1-20180928012546-x86_64-linux.hart
```
Comment (mgamini) [inline DEVELOPING.md]: *backline

File: components/builder-web/README.md [hunk 4]
```diff
@@ -4,40 +4,36 @@ This is the web application for Builder. It's a single-page application built wi
...
+```
 
-  * Set up a custom GitHub OAuth application to run the dev service somewhere other than `localhost:3000`. See below for instructions on how to do that.
+### Host OS Provisioning
 
-### Installing Node
+You will need to install `node`.
 
-We suggest using [NVM](https://github.com/creationix/nvm) (Node Version Manager) to install the version of Node specified in  [.nvmrc](.nvmrc). Follow [the instructions in the NVM docs](https://github.com/creationix/nvm#installation) to set that up.
+We suggest using [NVM](https://github.com/creationix/nvm) (Node Version Manager) to install the version of Node specified in [.nvmrc](.nvmrc). Follow [the instructions in the NVM docs](https://github.com/creationix/nvm#installation) to set that up.
```
Comment (mgamini) [inline components/builder-web/README.md]: (I prefer `n`, but maybe that's just me)

File: DEVELOPING.md (hunk 2 above)
Comment (chefsalim) [reply DEVELOPING.md]: direenv is installed by the provision.sh script so we should be ok

File: DEVELOPING.md (hunk 3 above)
Comment (chefsalim) [reply DEVELOPING.md]: good catch!

File: components/builder-web/README.md (hunk 4 above)
Comment (chefsalim) [reply components/builder-web/README.md]: Not clear, can you elaborate?

File: DEVELOPING.md (hunk 1 above)
Comment (chefsalim) [reply DEVELOPING.md]: Will add

Comment (chefsalim) [reply DEVELOPING.md]: Done - this section is updated with new NAT config info

File: DEVELOPING.md (hunk 3 above)
Comment (chefsalim) [reply DEVELOPING.md]: Fixed

**Pull request merged**